src/pip/_internal/commands/debug.py

```diff
@@ -134,10 +134,7 @@ def show_tags(options: Values) -> None:
 
 
 def ca_bundle_info(config: Configuration) -> str:
-    levels = set()
-    for key, _ in config.items():
-        levels.add(key.split(".")[0])
-
+    levels = {key.split(".")[0] for key, _ in config.items()}  # noqa: PERF102
```

Comment:
```suggestion
    # Ruff misidentifies config as a dict.
    # Configuration does not have support the mapping interface.
    levels = {key.split(".")[0] for key, _ in config.items()}  # noqa: PERF102
```

(Something like this, so future readers have an easier time understanding the noqa marker.